ui/src/app/views/account-creation/component.spec.ts

```diff
@@ -0,0 +1,194 @@
+        ClarityModule.forRoot()
+      ],
+      declarations: [
+        LoginComponent,
+        AccountCreationComponent,
+        AccountCreationSuccessComponent,
+        InvitationKeyComponent,
+        PageTemplateSignedOutComponent,
+        RoutingSpinnerComponent
+      ],
+      providers: [
+        { provide: LoginComponent, useValue: {}},
+        { provide: InvitationKeyComponent, useValue: {}},
+        { provide: ProfileService, useValue: profileServiceStub },
+        {
+          provide: ServerConfigService,
+          useValue: new ServerConfigServiceStub({
+            gsuiteDomain: 'fake-research-aou.org'
+          })
+        }]
+    }).compileComponents().then(() => {
+      fixture = TestBed.createComponent(AccountCreationComponent);
+      tick();
+    });
+    tick();
+  }));
+
+  it('handles selecting password', fakeAsync(() => {
+    updateAndTick(fixture);
+    expect(fixture.componentInstance.passwordOffFocus).toBeUndefined();
+    simulateEvent(fixture, fixture.debugElement.query(By.css('#password')), 'focus');
+    expect(fixture.componentInstance.passwordOffFocus).toBe(false);
+    simulateEvent(fixture, fixture.debugElement.query(By.css('#password')), 'blur');
+    expect(fixture.componentInstance.passwordOffFocus).toBe(true);
+  }));
+
+  it('handles each password failure case', fakeAsync(() => {
+    updateAndTick(fixture);
+    simulateEvent(fixture, fixture.debugElement.query(By.css('#password')), 'focus');
+    simulateInput(fixture, fixture.debugElement.query(By.css('#password')), 'password');
+    simulateEvent(fixture, fixture.debugElement.query(By.css('#password')), 'blur');
+    expect(fixture.componentInstance.containsLowerAndUpperError).toBeTruthy();
+    expect(fixture.componentInstance.isPasswordValidationError).toBeTruthy();
+    simulateInput(fixture, fixture.debugElement.query(By.css('#password')), 'Passwor');
+    expect(fixture.componentInstance.showPasswordLengthError).toBeTruthy();
+    expect(fixture.componentInstance.isPasswordValidationError).toBeTruthy();
+    simulateInput(fixture, fixture.debugElement.query(By.css('#password')),
+      'ThisissuchalongpasswordThisissuchalongpasswordThisissuchalongpasswordThisissuchalongpass \
+       ThisissuchalongpasswordThisissuchalongpasswordThisissuchalongpasswordThisissuchalongpass \
+       ThisissuchalongpasswordThisissuchalongpasswordThisissuchalongpasswordThisissuchalongpass');
+    expect(fixture.componentInstance.showPasswordLengthError).toBeTruthy();
+    expect(fixture.componentInstance.isPasswordValidationError).toBeTruthy();
+    simulateInput(fixture, fixture.debugElement.query(By.css('#password')), 'Password');
+    expect(fixture.componentInstance.showPasswordLengthError).toBeFalsy();
+    expect(fixture.componentInstance.containsLowerAndUpperError).toBeFalsy();
+    expect(fixture.componentInstance.isPasswordValidationError).toBeFalsy();
+  }));
+
+  it('only shows password errors when off focus', fakeAsync(() => {
```

Comment:
Would be nice if one of these test cases covered the bug I noticed (checkmark for empty password field). I don't think it's covered but I could be wrong.